pkg/aws/policies.go

```diff
@@ -1975,3 +1989,78 @@ func (c *awsClient) listRoleAttachedPolicies(roleName string) ([]iamtypes.Attach
 
 	return attachedPoliciesOutput.AttachedPolicies, nil
 }
+
+// check whether the policy contains specified tags
+func isPolicyTagFit(c client.IamApiClient, poilcyArn *string, tagFilter map[string]string) (bool, error) {
+	if len(tagFilter) != 0 {
+		tags, err := c.ListPolicyTags(context.Background(),
+			&iam.ListPolicyTagsInput{
+				PolicyArn: poilcyArn,
+			},
+		)
+		if err != nil {
+			return false, err
+		}
+		fitTagSize := 0
+		for _, tag := range tags.Tags {
+			value, ok := tagFilter[aws.ToString(tag.Key)]
+			if ok && value != aws.ToString(tag.Value) {
+				return false, nil
+			}
+			fitTagSize++
+		}
+		if fitTagSize < len(tagFilter) {
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+func getAttachedPolicies(c client.IamApiClient, role string,
+	tagFilter map[string]string) ([]string, error) {
+	policyArr := []string{}
+	policiesOutput, err := c.ListAttachedRolePolicies(context.Background(),
+		&iam.ListAttachedRolePoliciesInput{
+			RoleName: aws.String(role),
+		})
+	if err != nil && !awserr.IsNoSuchEntityException(err) {
+		return policyArr, err
+	}
+	for _, policy := range policiesOutput.AttachedPolicies {
+		tagFit, err := isPolicyTagFit(c, policy.PolicyArn, tagFilter)
+		if err != nil {
+			return policyArr, err
+		}
+		if tagFit {
+			policyArr = append(policyArr, aws.ToString(policy.PolicyArn))
+		}
+	}
+	return policyArr, nil
+}
+
+func getAcctRolePolicyTags(prefix string) map[string]string {
+	tagmap := map[string]string{}
+	tagmap[tags.RedHatManaged] = TrueString
+	tagmap[tags.RolePrefix] = prefix
+	return tagmap
+}
+
+func getOperatorRolePolicyTags(c client.IamApiClient, roleName string) (map[string]string, error) {
+	tagmap := map[string]string{}
+	tagmap[tags.RedHatManaged] = TrueString
+	roleTags, err := c.ListRoleTags(context.Background(), &iam.ListRoleTagsInput{
+		RoleName: aws.String(roleName),
+	})
+	if err != nil {
+		return tagmap, err
+	}
+	for _, tag := range roleTags.Tags {
+		switch aws.ToString(tag.Key) {
+		case tags.OperatorName:
+			tagmap[tags.OperatorName] = aws.ToString(tag.Value)
+		case tags.OperatorNamespace:
+			tagmap[tags.OperatorNamespace] = aws.ToString(tag.Value)
```

Comment:
I think this can be a single case:
```
key := aws.ToString(tag.Key)
switch key {
case tags.OperatorName, tags.OperatorNamespace:
    tagmap[key] = aws.ToString(tag.Value)
}
```

Reply:
changed